app/helpers/check_answers_helper.rb

```diff
@@ -1,154 +1,255 @@
-      # work to make them readable.
-
-      next if skip_question?(question)
-
-      next transport_type if question.eql?("transport_type")
-      next offer_care_qualifications if question.eql?("offer_care_qualifications")
-      next offer_staff_type if question.eql?("offer_staff_type")
-      next link_to_parent_page(question, "offer_staff_type") if question.eql?("offer_staff_charge")
-      next link_to_parent_page(question, "construction_services") if question.eql?("construction_services_other")
-      next link_to_parent_page(question, "construction_services") if question.eql?("construction_cost")
-      next link_to_parent_page(question, "it_services") if question.eql?("it_services_other")
-      next link_to_parent_page(question, "it_services") if question.eql?("it_cost")
-      next link_to_parent_page(question, "transport_type") if question.eql?("transport_cost")
-      next link_to_parent_page(question, "offer_care_qualifications") if question.eql?("care_cost")
-      next link_to_parent_page(question, "rooms_number") if question.eql?("accommodation_cost")
-      next link_to_parent_page(question, "offer_space_type") if question.eql?("space_cost")
-
-      value = concat_answer(session[question], question)
+  def sections
+    {
+      1 => "accommodation",
+      2 => "transport",
+      3 => "space",
+      4 => "staff",
+      5 => "care",
+      6 => "services",
+      7 => "other_support",
+      8 => "location",
+      9 => "business_details",
+      10 => "contact_details",
+    }.sort.to_h.values
+  end
 
+  def question_items(question)
+    if session[question].is_a?(Array)
       [{
-        field: t("coronavirus_form.questions.#{question}.title"),
-        value: sanitize(value),
-        edit: {
-          href: "#{question.dasherize}?change-answer",
-        },
+        field: I18n.t("coronavirus_form.questions.#{question}.title"),
+        value: render("coronavirus_form/check_your_answers/helpers/list", items: session[question]),
+      }]
+    else
+      [{
+        field: I18n.t("coronavirus_form.questions.#{question}.title"),
+        value: session[question],
       }]
     end
+  end
 
-    items.compact.flatten
+  def counts_key(name)
+    "#{name.downcase.gsub(' ', '_')}_number".to_sym
   end
 
-  def skip_question?(question)
-    question.in?(SKIPPABLE_QUESTIONS) && session[question].blank?
+  def humanize_count(count, postfix)
+    number = number_with_delimiter(count, delimiter: ",")
+    pluralize(number, postfix)
   end
 
-  def transport_type
-    [{
-      field: t("coronavirus_form.questions.transport_type.title"),
-      value: sanitize(transport_type_info),
-      edit: {
-        href: "transport-type?change-answer",
+  def create_list_with_named_values(names, counts, postfix)
+    return "" if names.blank?
+
+    list = []
+    names.each do |name|
+      list << "#{name} (#{humanize_count(counts[counts_key(name)], postfix)})"
+    end
```

Comment:
I'd suggest using `map` here, rather than creating an empty array and pushing into it within an `each` iteration.
```ruby
list = names.map do |name|
  "#{name} (#{humanize_count(counts[counts_key(name)], postfix)})"
end
```

Reply:
Thanks @reggieb 👍  Good spot - we will fix this.